### ISSUE TITLE
GraphQL logOut query dose not work 

### ISSUE THREAD

**Issue by Vildnex** [parse-community/parse-server#8257]
### New Issue Checklist
- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the [latest version of Parse Server](https://github.com/parse-community/parse-server/releases). 

### Issue Description
I am trying to do a logout of a user via a GraphQL query, but is not working as expected.

### Steps to reproduce
After reading the documentation from [here](https://www.back4app.com/docs/parse-graphql/graphql-login) it showed that the query for login is like this:

```
mutation Login($username: String!="a", $password: String!="b") {
    logIn(input: {
        username: $username
        password: $password
    }){
        viewer{
            sessionToken
        }
    }
}

```
Afterward, the user will be login on to the server and I will have a new session added:

![image](https://user-images.githubusercontent.com/10352059/197402374-80540e68-aa04-4d24-8c4c-2b654014d257.png)

But my problem arises when I have when I am trying to do a logout query. So based also on this [documentation](https://www.back4app.com/docs/parse-graphql/graphql-logout-mutation) the logout query should look something like this:

```
mutation Logout($clientId: String!) {
    logOut(input: { clientMutationId: $clientId}) {
        ok
    }
}
```

So far as I understood the **clientMutationId** should be the objectID from the user (I am not 100% sure), so I've tried the following 

- **ObjectID of Session:** r5zqQngLHT 
- **ObjectID of User:** dCv5FiGtRw
- **Session Token:** r:8b8bdde1b8a5314a5a90ae8a47639864

None of the inputs worked and in all of the cases the session of my user remained open, and in all of the cases the output from this query was like this:

```
{
  "data": {
    "logOut": {
      "ok": true
    }
  }
}
```
Even in the case where clientMutationId is empty "". Did I understand it wrong? How should I log out of my user GraphQL if the version is not working?

GraphQL

### Environment
**I use a docker-compose.yaml**
```
version: '3.9'

services:
  database:
    image: mongo:6.0.2
    environment:
      MONGO_INITDB_ROOT_USERNAME: admin
      MONGO_INITDB_ROOT_PASSWORD: admin
    volumes:
      - ${HOME}/_DOCKER_DATA_/database:/data/db

  server:
    restart: always
    image: parseplatform/parse-server:5.2.8
    ports:
      - 1337:1337
    environment:
      - PARSE_SERVER_APPLICATION_ID=COOK_APP
      - PARSE_SERVER_APPLICATION_NAME=COOK_NAME
      - PARSE_SERVER_MASTER_KEY=MASTER_KEY_1
      - PARSE_SERVER_CLIENT_KEY=CLIENT_KEY_1
      - PARSE_SERVER_DATABASE_URI=mongodb://admin:admin@mongo/parse_server?authSource=admin
      - PARSE_SERVER_URL=http://10.0.2.2:1337/parse
      - PARSE_SERVER_MOUNT_GRAPHQL=true
    links:
      - database:mongo
    volumes:
      - ${HOME}/_DOCKER_DATA_/server:/data/server

  dashboard:
    image: parseplatform/parse-dashboard:5.0.0-beta.1
    ports:
      - "4040:4040"
    depends_on:
      - server
    environment:
      - PARSE_DASHBOARD_APP_ID=COOK_APP
      - PARSE_DASHBOARD_MASTER_KEY=MASTER_KEY_1
      - PARSE_DASHBOARD_USER_ID=admin
      - PARSE_DASHBOARD_USER_PASSWORD=admin
      - PARSE_DASHBOARD_ALLOW_INSECURE_HTTP=true
      - PARSE_DASHBOARD_SERVER_URL=http://localhost:1337/parse
      - PARSE_DASHBOARD_GRAPHQL_SERVER_URL=http://localhost:1337/graphql
    volumes:
      - ${HOME}/_DOCKER_DATA_/dashboard:/data/dashboard

```

Am I missing something?

**Comment by mtrezza** [comment]
There is a test for logout:

https://github.com/parse-community/parse-server/blob/fb503320c3c48f47c4ec9124c5b007af242c1e1f/spec/ParseGraphQLServer.spec.js#L7186

Could you take a look and maybe open a PR with a failing test that demonstrates the issue?

**Comment by Vildnex** [comment]
Sure, but how can I run this test in order to check?

**Comment by mtrezza** [comment]
To run the existing test you change the `it` to `fit` to only run that specific test. Then run `npm test spec/ParseGraphQLServer.spec.js`.

**Comment by Vildnex** [comment]
Ok then, I will test it tomorrow and I will let you know

**Comment by Vildnex** [comment]
Ok so... I'm still not sure how to run that test, but I download that js file you give to me and I run that command. After that I received this error:
```

npm ERR! code ENOENT
npm ERR! syscall open
npm ERR! path /home/vlad/Documents/Test_GraphQL/package.json
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, open '/home/vlad/Documents/Test_GraphQL/package.json'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent 

npm ERR! A complete log of this run can be found in the:
npm ERR!     /home/vlad/.npm/_logs/2022-10-23T20_24_17_153Z-debug-0.log
```

After that, I also added the package.json, and I run it the command back and I get this:


```
> parse-server@5.3.0-alpha.30 pretest
> cross-env MONGODB_VERSION=${MONGODB_VERSION:=5.3.2} MONGODB_TOPOLOGY=${MONGODB_TOPOLOGY:=standalone} MONGODB_STORAGE_ENGINE=${MONGODB_STORAGE_ENGINE:=wiredTiger} mongodb-runner start

sh: 1: cross-env: not found
```

After this, I cloned the whole project from git via:

```
git clone https://github.com/parse-community/parse-server
cd parse-server
git checkout -b fb503320c3c48f47c4ec9124c5b007af242c1e1f
#replace the `it` with `fit` from `ParseGraphQLServer.spec.js`
npm test spec/ParseGraphQLServer.spec.js
```

And I get the same error as about, how exactly should I run the test?

**Comment by mtrezza** [comment]
The test is not for you to run it against your application. The test runs as part of the "development code" within Parse Server. In other words, Parse Server is testing itself. These tests run every time before a new Parse Server version is released. You can start the test by cloning the `parse-server` repo locally, then run `npm i && npm test`. 

I mentioned the test for you to compare the code in the test to the code you are using in your app. The test code works, otherwise Parse Server would fail when testing against itself.

**Comment by mtrezza** [comment]
cc @Moumouls

**Comment by Moumouls** [comment]
HI @mtrezza  @Vildnex , the `logOut` mutation works correctly.

The client mutation ID, is a special field ONLY used and available in GraphQL Relay Spec to track operations into a client app store.

Logout mutation ONLY logout the currently connected user. Parse Server uses the session token provided by the GraphQL mutation (in the HTTP headers) to perform the logout.

If you want to log out a specific User, you need to find and delete the or all session tokens used by the User.

You can also find the session token, create a GraphQL request with the found session token in the headers to run the graphql query as the targeted user and perform the logout.

**Comment by Vildnex** [comment]
Hello,

Ok so let me see that I get it, the `logOut` is going to the logout operation internally and is going to keep my whole session as previously intact. So in order to properly logout a user I should make a special query that will delete that session row from the _Session table, right? But if this is like this I do have 2 questions:

1. Do I have to use the `logOut` function at all then? Because if I will manually delete the entries from _Session anyway, isn't it the whole point?
2. I've just checked and in the same manner as before I've tried to `logIn` a user multiple times and it worked every single time with no errors. Every single time when I run the 'logIn' function it will be added a new row to _Session. Should not be the case that when I am trying to do `logIn` with the same creditless to get an error that I am already `logIn` or something?
